### PR TITLE
Remove superflous T_ASS_LVAR_xx

### PR DESCRIPTION
**Pull request by markuspf**
As far as I can tell no code produced expressions with this type. Lets see what travis has to say.

*Edit:* It is not true that no code was produced, I actually removed the part of the coder that produced this code. I will post some benchmarks below that show that these changes do not seem to affect perfomance negatively.

### CODE REVIEW

**Comment by fingolfin** [comment]
See also PR #1108 resp. commit 8d2ca938c2f4102208b72e21a87d723f9d8446e0

Though I see you closed this now... Huh?

File: src/code.c [hunk 1]
```diff
@@ -2257,12 +2257,7 @@ void CodeAssLVar (
     Expr                rhsx;           /* right hand side expression      */
 
     /* allocate the assignment                                             */
-    if ( lvar <= 16 ) {
-        ass = NewStat( T_ASS_LVAR + lvar, 2 * sizeof(Stat) );
-    }
-    else {
-        ass = NewStat( T_ASS_LVAR,        2 * sizeof(Stat) );
-    }
```
Comment (fingolfin) [inline src/code.c]: Hmmm... so `T_ASS_LVAR_xx` is/are used after all -- apparently as an optimization, and this PR removes the optimization.

So, did you perform any benchmarking to see how this affects things?

File: src/vars.c [hunk 2]
```diff
@@ -2953,30 +2953,10 @@ static Int InitKernel (
 
     /* install executors, evaluators, and printers for local variables     */
     InstallExecStatFunc( T_ASS_LVAR       , ExecAssLVar);
-    InstallExecStatFunc( T_ASS_LVAR_01    , ExecAssLVar01);
```
Comment (fingolfin) [inline src/vars.c]: This is the only place which references `ExecAssLVar01` etc., so this PR should also remove it and its siblings.

File: src/code.c (hunk 1 above)
Comment (markuspf) [inline src/code.c]: I ran the following snippets (repeatedly, compared average running times)

```
G := Group(AtlasGenerators("Fi24", 1).generators);;
gap> t := NanosecondsSinceEpoch();; Size(G);; t := NanosecondsSinceEpoch() - t;
```

```
G := Group(AtlasGenerators("Fi23", 1).generators);;
gap> t := NanosecondsSinceEpoch();; Size(G);; t := NanosecondsSinceEpoch() - t;
```

```
f := function()
    local t,i,
          r0,r1,r2,r3,r4,r5,r6,r7,
          r8,r9,r10,r11,r12,r13,r14,r15,
          r16,r17,r18,r19,r20,r21,r22,r23,r24;

    t := NanosecondsSinceEpoch();
    for i in [1..1000000] do
        r0 := Random([1..100]);
        r1 := r0 + r0;
    od;
    t := NanosecondsSinceEpoch() - t;
    Print("t1: ", t, "\n");

    t := NanosecondsSinceEpoch();
    for i in [1..1000000] do
        r24 := Random([1..100]);
        r23 := r24 + r24;
    od;
    t := NanosecondsSinceEpoch() - t;
    Print("t2: ", t, "\n");
end;
f();
```

Without any measurable performance impact. If anyone has any more suggestions for benchmarks, I am all ears, also feel free to run this on your machine to cross-check.

**Comment by fingolfin** [comment]
Great then!

For the record, this code already appears in the very first commit in the old repository:
```
commit 03889246749445b1480e2b17a84df7f41cd550f3
Author: Martin Schönert <martin.schoenert@math.rwth-aachen.de>
Date:   1996-07-03 14:55:06 +0000

    initial revision under CVS
```

**Comment by stevelinton** [comment]
My suspicion is that this code may have given a performance boost on systems 20 years ago, since it meant fewer memory accesses and less computing for an assignment to a local variable. Now however it's still just one cache line and that dominates everything else. So let's simplify the code.

**Comment by stevelinton** [comment]
The corresponding T_REF_LVAR tnums have been removed some time ago it seems. You can still see the gap they used to occcupy in the numbering of expression types.

**Comment by fingolfin** [comment]
@stevelinton yeah, that was done in the PR #1108 resp. commit 8d2ca I mentioned above.

File: src/code.c (hunk 1 above)
Comment (ChrisJefferson) [inline src/code.c]: I also checked I can't get this to go and faster. it's not suprising, as you the T_ASS_LVAR_xx aren't even any smaller, they still contain the lvar value, so all you are saving is reading one of two values which are right next to each other, which are almost certainly in the same cache line so the CPU already has them both anyway.

**Comment by ChrisJefferson** [comment]
I am happy with this, but will also leave it for a while just in case @frankluebeck or anyone else wants to comment, as it is a fairly deep change.

**Comment by fingolfin** [comment]
Talked to Frank about it, he has no objections.

**Pull request merged**